app/editor/components/MentionMenu.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { v4 } from "uuid";
+import { MenuItem } from "@shared/editor/types";
+import useRequest from "~/hooks/useRequest";
+import useStores from "~/hooks/useStores";
+import CommandMenu, { Props } from "./CommandMenu";
+import MentionMenuItem from "./MentionMenuItem";
+
+interface MentionItem extends MenuItem {
+  name: string;
+  title: string;
+  attrs: {
+    "data-type": string;
```

Comment:
```suggestion
    type: MentionType;
```

Lets make it an enum `MentionType.User` is the only option for now